modules/claims_api/app/services/claims_api/poa_document_service.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class PoaDocumentService < DocumentServiceBase
+    LOG_TAG = 'Poa_Document_service'
+
+    def create_upload(poa:, pdf_path:, doc_type:, action:)
+      unless File.exist? pdf_path
+        ClaimsApi::Logger.log(LOG_TAG, detail: "Error creating upload doc: #{pdf_path} doesn't exist,
+                                                    poa_id: #{poa.id}")
+        raise Errno::ENOENT, pdf_path
+      end
+
+      body = generate_body(poa:, doc_type:, pdf_path:, action:)
+      doc_type_name = 'POA'
+      ClaimsApi::BD.new.upload_document(identifier: poa.id, doc_type_name:, body:)
+    end
+
+    private
+
+    ##
+    # Generate form body to upload a document
+    #
+    # @return {parameters, file}
+    def generate_body(poa:, doc_type:, pdf_path:, action:)
+      auth_headers = poa.auth_headers
+      veteran_name = compact_veteran_name(auth_headers['va_eauth_firstName'],
+                                          auth_headers['va_eauth_lastName'])
+      ptcpnt_vet_id = auth_headers['va_eauth_pid']
+      participant_id = find_ptcpnt_vet_id(auth_headers, ptcpnt_vet_id)
```

Comment:
Slightly confused - one would expect `ptcpnt_vet_id = find_ptcpnt_vet_id(auth_headers, ptcpnt_vet_id)`, right? What is the difference between `ptcpnt_vet_id` and `participant_id`?

Reply:
Interesting, I wonder if it would work if they were swapped. Anyway, seems to be working.